src/v2i-hub/CDASimAdapter/src/CDASimConnection.cpp

```diff
@@ -37,11 +37,55 @@ namespace CDASimAdapter{
         PLOG(logINFO) << "CARMA-Simulation connection is successful!" << std::endl;
         return _connected;
     }
-    bool CDASimConnection::carma_simulation_handshake(const std::string &simulation_ip, const uint simulation_registration_port, 
+    bool CDASimConnection::carma_simulation_handshake(const std::string &simulation_ip, const uint infrastructure_id, const uint simulation_registration_port, 
                                 const std::string &local_ip,  const uint time_sync_port, const uint v2x_port, 
-                                const WGS84Point &location) {
-        // TODO: Standup UDP Server and Client for registration and implement handshake
-        return false;                                
+                                const WGS84Point &location) 
+    {
+        
+        //Get remote endpoint
+        try
+        {
+            remote_udp_ep_ = boost::asio::ip::udp::endpoint(boost::asio::ip::address::from_string(simulation_ip), simulation_registration_port);
+        }
+        catch(std::exception e)
+        {
+            PLOG(logERROR) << "Encountered runtime error when resolving remote udp end point given: " << e.what() << std::endl;
+            throw e;
+        }
+
+        io_.reset(new boost::asio::io_service());
+        udp_out_socket_.reset(new boost::asio::ip::udp::socket(*io_,remote_udp_ep_.protocol()));
+
+        work_.reset(new boost::asio::io_service::work(*io_));
+
+         // Create JSON message with the content 
+        Json::Value message;   
+
+        message["rxMessageIpAddress"] = local_ip;
+        message["infrastructureId"] = infrastructure_id;
+        message["rxMessagePort"] = v2x_port;
+        message["timeSyncPort"] = time_sync_port;
+        
+        message["location"]["latitude"] = location.Latitude;
+        message["location"]["longitude"] = location.Longitude;
+        message["location"]["elevation"] = location.Elevation;
+ 
+        try
+        {
+            Json::StyledWriter writer;
+            std::string message_str = writer.write(message);
+            udp_out_socket_->send_to(boost::asio::buffer(message_str), remote_udp_ep_);
```

Comment:
Is there a reason you did not use the UdpClient class?

Reply:
No reason, I just used the carma-platform's ns3 adapter as a reference. I will replace it right now